File: 2ndsem/DS/gcd.c
#include <stdio.h>
int gcd(int a, int b) {
if (b == 0) {
return a;
}
return gcd(b, a % b);
}
int main(void) {
int a, b;
printf("Enter two integers: ");
scanf("%d %d", &a, &b);
printf("The greatest common divisor of %d and %d is %d.\n", a, b, gcd(a, b));
return 0;
}

File: 3rd Sem/DSA/gcd.java
class gcd {
    static int gcdImplement(int a, int b) {
        if (b == 0)
            return a;
        else
            return gcdImplement(b, a % b);
    }

    public static void main(String[] args) {
        int a = 12;
        int b = 24;

        int gcd = gcdImplement(a, b);
        System.out.println("The GCD of " + a + " & " + b + " = " + gcd);
    }
}